### PR TITLE
show the information about the analysis server's obs. port

### PR DESCRIPTION
**Pull request by devoncarew**
- show the information about the analysis server's obs. port
- make the analysis server debug option visible on the settings page

screen shot 2016-06-04 at 3 18 34 pm

@danrubel (likely will TBR)

### CODE REVIEW

**Comment by devoncarew** [comment]
(to help with https://github.com/flutter/flutter/issues/4384)

File: lib/analysis/analysis_server_lib.dart [hunk 1]
```diff
@@ -81,14 +81,14 @@ class Server {
...
 
-    try {
-      _onReceive.add(message);
+    if (!message.startsWith('{')) {
+      _logger.warning('unknown message: ${message}');
+      return;
+    }
 
+    try {
       var json = JSON.decode(message);
```
Comment (danrubel) [inline lib/analysis/analysis_server_lib.dart]: try/catch around JSON.decode ?

File: lib/analysis_server.dart [hunk 2]
```diff
@@ -126,6 +126,11 @@ class AnalysisServer implements Disposable {
     });
 
     onReceive.listen((String message) {
+      if (message.startsWith('Observatory listening') ||
+          message.startsWith('Observatory no longer listening')) {
+        atom.notifications.addInfo('Analysis server', detail: message.trim(), dismissable: true);
```
Comment (danrubel) [inline lib/analysis_server.dart]: Remove ```dismissable: true``` so that message automatically is removed after a while?

**Comment by danrubel** [comment]
LGTM

File: lib/analysis_server.dart (hunk 2 above)
Comment (devoncarew) [inline lib/analysis_server.dart]: This message will contain the port to connect to for the observatory connection, so we'll want it to persist for a while.

**Pull request merged**